src/jvm/main/org/jetbrains/kotlinx/lincheck/Utils.kt

```diff
@@ -230,6 +232,83 @@ internal val Class<*>.allDeclaredFieldWithSuperclasses get(): List<Field> {
     return fields
 }
 
+/**
+ * Traverses all fields of a given object in bfs order.
+ *
+ * In case [treatArrayElementsAsFields] is
+ * set to `false`, then array fields will be traversed instead of elements.
+ *
+ * The [callback] is called for each tuple of `(fieldOwner, field, fieldValue)` of the [obj],
+ * the return value of the callback determines, what object should be traversed further
+ * (it can be something different from `fieldValue`).
+ *
+ * Parameters of [callback]:
+ * - `fieldOwner`: either `Any` - some class instance that owns the field, or `Class<*>` is case if `field` is static.
```

Comment:
Isn't there a semantics collision when some `Class<*>` instance is passed as object, and its non-static field is being inspected?

Reply:
Or, event worse, when a static field of `Class<*>` object is being inspected?

Reply:
I will change the description in javadoc, but it seems that traversing `Class<*>` is not needed in any of usage of this utility function in lincheck. In snapshot replaying I also will not need to traverse `Class<*>`, because I am starting a traverse of static variables by exact reference to their instances, and not via `Class<*>` of the declaring class.

Reply:
Let's then explicitly skip any `Class` objects in the traversal function.